docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+The command: 
+    
+    list appt
+    
+will list all the appointments in the `AppointmentMap`. 
+
+**Implementation** 
+
+![List Appointment Sequence Diagram](images/ListAppointmentSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `list appt`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will create an object `ListAppointmentCommand`.
+
+4. `HappyPills` then calls `ListAppointmentCommand.execute()` to execute the method. 
+
+5. In `ListAppointmentCommand.execute()`, it calls `TextUi.getAppointmentList()` which is then returned as the display message. 
+
+#### 4.3.6. Find Appointments of Patient
+
+The user is able to find all the appointments a specific patient has.
+The command: 
+    
+    find appt S7777777Z
+    
+will list all the appointments that the patient with NRIC S7777777Z has. 
+
+**Implementation**
+
+![Find Appointment Sequence Diagram](images/FindAppointmentSequenceDiagram.jpg)
+
+
+### 4.4. Storage
+
+This is an internal feature of the program, implemented to allow users to recover information even after HappyPills is 
+closed in the terminal. This is achieved by storing all relevant information in a text file using a structured format.
+
+**Implementation** 
+
+ The current methods implemented in this class and a brief description of each method:
+ - `writeAllToFile` — writes the entire list of item to the specified text file.
+ - `addSingleItemToFile` — adds a new item as a single string to the end of specified text file.
+ - `loadPatientFromFile` — access the patient file and retrieve all information in the file as strings.
+ - `parsePatientFileContent` — process line-by-line to create a patient object and add to the shared patient map.
+ - `loadAppointmentFromFile` — access the patient file and retrieve all information in the file as strings.
+ - `parseAppointmentFileContent` — process line-by-line to create an appointment object and add to the 
+                                   shared appointment map.
+ 
+ The following diagram shows how each command interacts with the other classes.
+ 
+ *writeAllToFile*
+  
+  This storage method : `writeAllToFile`, gets a single formatted string from TextUi, containing details of all the 
+  objects in a map. Each class object would have a class method to construct a toSave() string, and 
+  each object is stored as a single line within the text file. Hence, the toSave() string is constructed with  
+  '|' as a divider, and a newline to indicate the end of the object. The formatted string is a concatenation of all 
+  the toSave() strings of the objects in the list. `writeAllToFile` then writes the entire string into the text file, 
+  overwriting any existing strings in the file.
+  This is implemented for edit and delete commands as they cannot be appended.
+  
+  ![writing](images/StorageWriteAll.png)
```

Comment:
Perhaps you could use the same software you used for the previous sequence diagrams to create this diagram as well? This way, all the sequence diagrams in the guide will look similar and it will have a more professional feel.